### DISCUSSION TITLE
Cache Attribute

### DISCUSSION THREAD

**Discussion by vmiguellima** [laravel/framework#41834] [Q&A]
I've created an accessor to retrieve the `body` column in html. 

The attribute is working has expected the only thing that is not working, is caching.

```php
 /**
     * Get the body markdown in html
     *
     * @return \Illuminate\Database\Eloquent\Casts\Attribute
     */
    protected function html(): Attribute
    {            
        $mdParser = new \App\Markdown\SimpleMarkdown;

        return Attribute::make(
            get: fn () => $mdParser->parse($this->body)
        )->shouldCache();
    }
```

Then I call `$article->html` where needed.

But the subsequent calls take as long as the first one.

 I'm I doing something wrong? Should I use `Cache::remember` instead? Like this:

```php
  protected function html(): Attribute
  {            
      return Attribute::make(
          get: fn () => Cache::remember("article.$this->uuid.html", 60, function () {
              $mdParser = new \App\Markdown\SimpleMarkdown;
              return $mdParser->parse($this->body);
          })
      );
  }
```

Thanks in advance.

**Comment by louis-l** [comment]
@vmiguellima and for those who are looking to find the answer of what `shouldCache()` is actually doing:

I have set up a quick test:

```php
public function foo(): Attribute
{
     logger()->debug('here 111');
        
    return Attribute::get(function () {
        logger()->debug('here 222');
        return true;
    });
}
```

And when I call `foo` 3 times, here is the log output:

```
[2022-10-19 22:09:54] local.DEBUG: here 111  
[2022-10-19 22:09:54] local.DEBUG: here 111  
[2022-10-19 22:09:54] local.DEBUG: here 222  
[2022-10-19 22:09:54] local.DEBUG: here 111  
[2022-10-19 22:09:54] local.DEBUG: here 222  
[2022-10-19 22:09:54] local.DEBUG: here 111  
[2022-10-19 22:09:54] local.DEBUG: here 222  
```

Then, if I append `shouldCache()`, here is the output:

```
[2022-10-19 22:09:58] local.DEBUG: here 111  
[2022-10-19 22:09:58] local.DEBUG: here 111  
[2022-10-19 22:09:58] local.DEBUG: here 222  
[2022-10-19 22:09:58] local.DEBUG: here 111  
[2022-10-19 22:09:58] local.DEBUG: here 111  
```

**TL;DR**

The `shouldCache()` is like "in memory" cache, it will not persist and remember the computed value in future requests.

So I guess if you want to cache the value between different requests, you should go with `Cache::remember()` instead.
